### PR TITLE
RESFRAME-2729 - get cert paths for SSL requests

### PR DESCRIPTION
**Pull request by james-bowers**
related but already merged PRs:
https://github.com/bbc/mozart-fetcher-build/pull/14
https://github.com/bbc/mozart-fetcher-build/pull/15

Tested by doing:
`cat james-debug-fetcher-config.json | curl -X POST -d @- http://localhost:8080/collect` with a config that hits internal bbc endpoints and gets a response

### CODE REVIEW

File: lib/mozart_fetcher.ex [hunk 1]
```diff
@@ -21,8 +21,23 @@ defmodule MozartFetcher do
     Application.get_env(:mozart_fetcher, :max_connections)
   end
 
-  def cert do
-    System.get_env("DEV_CERT_PEM")
+  def request_ssl do
```
Comment (ettomatic) [inline lib/mozart_fetcher.ex]: maybe a test for this?

Comment (james-bowers) [reply lib/mozart_fetcher.ex]: done :)

File: lib/mozart_fetcher.ex [hunk 2]
```diff
@@ -21,8 +21,23 @@ defmodule MozartFetcher do
...
+    case environment() do
+      :prod -> [certfile: cert_path(), cacertfile: ca_cert_path(), keyfile: key_file_path()]
+      _ -> [certfile: System.get_env("DEV_CERT_PEM")]
+    end
+  end
+
+  defp cert_path do
+    System.get_env("cert_file_path")
```
Comment (samfrench) [inline lib/mozart_fetcher.ex]: Should we use the same convention for our environment variables, by uppercasing them all?

File: lib/mozart_fetcher.ex [hunk 3]
```diff
@@ -21,8 +21,23 @@ defmodule MozartFetcher do
...
+  end
+
+  defp ca_cert_path do
+    System.get_env("ca_file_path")
+  end
+
+  defp key_file_path do
+    System.get_env("key_file_path")
```
Comment (samfrench) [inline lib/mozart_fetcher.ex]: Are we keeping the same naming we use in requester, such as key_location?

Comment (james-bowers) [reply lib/mozart_fetcher.ex]: So I've intentionally not used requester's naming as I didn't think the naming for that was very good, as it switched between ending `_file` and `_location`. Now they all end in the same format, `_file_path`

File: lib/mozart_fetcher.ex (hunk 2 above)
Comment (james-bowers) [reply lib/mozart_fetcher.ex]: can do, we didn't for requester, and these environment variables are initially set in the cosmos config, and the convention there is for the names to be lower case.

Comment (woodyblah) [reply lib/mozart_fetcher.ex]: Maybe we should use capital letters for the cosmos config keys? https://cosmos.tools.bbc.co.uk/services/ingress/test/configuration we have INSTANCE_ROLE_NAME but also `td_ingest_bucket` which is required to be lower case by the td agent :(

Comment (james-bowers) [reply lib/mozart_fetcher.ex]: yeah that'd be the way to do it, but I'm not fussed either, I think I'll merge this

**Pull request merged**